runtime/doc/lsp.txt

```diff
@@ -946,59 +966,61 @@ apply_text_edits({text_edits}, {bufnr})
 apply_workspace_edit({workspace_edit})
                 TODO: Documentation
 
-                                         *vim.lsp.util.diagnostics_by_buf*
-diagnostics_by_buf
-                A table containing diagnostics grouped by buf.
-
-                {<bufnr>: {diagnostics}}
-
-		{diagnostics} is an array of diagnostics.
-
-                By default this is populated by the
-                `textDocument/publishDiagnostics` callback via
-                |vim.lsp.util.buf_diagnostics_save_positions|.
-
-                It contains entries for active buffers. Once a buffer is
-                detached the entries for it are discarded.
-
 buf_clear_diagnostics({bufnr})          *vim.lsp.util.buf_clear_diagnostics()*
                 TODO: Documentation
- 
-                                        *vim.lsp.util.buf_diagnostics_count()*
-buf_diagnostics_count({kind})
-                Returns the number of diagnostics of given kind for current buffer.
-                Useful for showing diagnostics counts in statusline. eg:
 
+buf_clear_references({bufnr})            *vim.lsp.util.buf_clear_references()*
+                TODO: Documentation
+
+buf_diagnostics_count({kind})           *vim.lsp.util.buf_diagnostics_count()*
+                Returns the number of diagnostics of given kind for current
+                buffer.
+
+                Useful for showing diagnostic counts in statusline. eg:
 >
+
     function! LspStatus() abort
         let sl = ''
         if luaeval('vim.lsp.buf.server_ready()')
```

Comment:
TODO: This should get changed to something that doesn't sent server notifications.